File: backend/api/routes/game.js
const express = require('express');
const router = express.Router();
const { 
  getGames, 
  getGameById, 
  getGameBySlug,
  createGame,
  updateGame,
  deleteGame
} = require('../controllers/gameController');
const { uploadGame } = require('../middleware/uploadMiddleware');
const { authMiddleware, roleMiddleware } = require('../middleware');

// Public routes
router.get('/', getGames);
router.get('/id/:id', getGameById);
router.get('/:slug', getGameBySlug);

// Admin routes
router.post('/', authMiddleware, roleMiddleware('admin'), uploadGame, createGame);
router.put('/:id', authMiddleware, roleMiddleware('admin'), uploadGame, updateGame);
router.patch('/:id', authMiddleware, roleMiddleware('admin'), uploadGame, updateGame);
router.delete('/:id', authMiddleware, roleMiddleware('admin'), deleteGame);

module.exports = router;

File: fastapi/src/routes/game.py
from fastapi import APIRouter, Query
from src.controllers.game_controller import get_games, get_game_by_id, get_game_by_slug

router = APIRouter()


@router.get("/", response_model=dict)
async def get_games_route(category: str = Query(None)):
    return await get_games(category)


@router.get("/id/{game_id}", response_model=dict)
async def get_game_route(game_id: str):
    return await get_game_by_id(game_id)


@router.get("/{slug}", response_model=dict)
async def get_game_slug_route(slug: str):
    return await get_game_by_slug(slug)